src/RequestHandlerSwooleRunner.php

```diff
@@ -194,23 +207,98 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
         }
 
         $this->logger = $logger ?: new StdoutLogger();
+        $this->pidManager = $pidManager;
     }
 
     /**
      * Run the application
      */
     public function run() : void
     {
-        $this->swooleHttpServer->on('start', [$this, 'onStart']);
-        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
-        $this->swooleHttpServer->start();
+        $opts = new Getopt([
+            'd|daemonize'  => 'Daemonize the swoole server process',
+            'n|worker_num|num_worker=i' => 'The number of the worker process.',
+        ]);
+        $args = $opts->getArguments();
+        $action = $args[0] ?? null;
+        switch ($action) {
+            case 'stop':
+                $this->stopServer();
+                break;
+            case 'start':
+            default:
+                $this->startServer([
+                    'daemonize' => $opts->getOption('daemonize') ? (bool) $opts->getOption('daemonize') : false,
+                    'worker_num' => $opts->getOption('worker_num') ? (int) $opts->getOption('worker_num') : 1,
+                ]);
+                break;
```

Comment:
Note to self: I'm thinking we could abstract this functionality, and inject it into the runner. The logic here would then look like:

```php
switch ($this->console->getAction()) {
/* ... */
}
```

This would allow us to de-couple from zend-console, and potentially provide implementations both for `Zend\Console\Getopt` and symfony/console.